handlers/_swagger.js

```diff
@@ -319,31 +319,32 @@ function registerRoute(app, auth, additionalMiddleware, method, path, data, allo
             setDefaultQueryParams(req, data, logger);
             setDefaultHeaders(req, data, logger);
 
-            //Wrap the set function, which is responsible for setting headers
-            if (allowedTypes) {
-                //Validate that the content-type is correct per the swagger definition
-                wrapCall(res, 'set', function (name, value) {
-                    if (name === 'Content-Type') {
-                        var type = value.split(';')[0]; //parse off the optional encoding
-                        if (!_.contains(allowedTypes, type)) {
-                            logger.warn('Invalid content type specified: %s. Expecting one of %s', type, allowedTypes);
-                        }
+            // Wrap the set function, which is responsible for setting headers
+            var type = '';
+            // Validate that the content-type is correct per the swagger definition
+            wrapCall(res, 'set', function (name, value) {
+                if (name === 'Content-Type') {
+                    type = value.split(';')[0]; //parse off the optional encoding
+                    if (allowedTypes && !_.contains(allowedTypes, type)) {
+                        logger.warn('Invalid content type specified: %s. Expecting one of %s', type, allowedTypes);
                     }
-                });
-            }
+                }
+            });
 
             if (responseModelValidationLevel) {
                 var responseSender = res.send;
                 res.send = function (body) {
                     var isBodyValid = isValidDataType(body);
```

Comment:
it looks like it might be good to refactor this function's name, maybe to `isValidJsonData`

With that change, it may make sense to move the check for `'application/json'` to the outermost extent to protect the subsequent code. However, I am not sure how that will affect users who just call `res.send` without explicitly setting the `Content-Type` ...